mlflow/models/evaluation/base.py

```diff
@@ -151,6 +156,194 @@ def __str__(self):
+                ),
+                Parameter(
+                    "metrics",
+                    Parameter.KEYWORD_ONLY,
+                    annotation=Optional[Dict[str, MetricValue]],
+                    default=None,
+                ),
+                Parameter(
+                    "return_only_scores",
+                    Parameter.KEYWORD_ONLY,
+                    annotation=bool,
+                    default=False,
+                ),
+                *[
+                    Parameter(name, Parameter.KEYWORD_ONLY, annotation=Union[pd.Series, str, list])
+                    for name in allowed_kwargs_names
+                ],
+            ]
+        )
+        call_method = genai_call_method
+
+    else:
+
+        def _call_method(
+            self,
+            *,
+            return_only_scores: bool = False,
+            **kwargs,
+        ) -> Union[MetricValue, List[str], List[float]]:
+            """
+            Evaluate the metric on the given inputs and predictions.
+            Note: only keyword arguments are supported.
+
+            Args:
+                return_only_scores: If True, return only the scores from the metric.
+                    Otherwise, return the full MetricValue object. Default to False.
+                kwargs: additional arguments used to compute the metric.
+
+            Returns:
+                If return_only_scores is True, return the scores from the metric.
+                Otherwise, return the full MetricValue object.
+            """
+            result = self.eval_fn(**kwargs)
+            if return_only_scores:
+                return result.scores
+            return result
+
+        allowed_kwargs_params = inspect.signature(eval_fn).parameters
+        _call_method.__signature__ = Signature(
+            parameters=[
+                Parameter("self", Parameter.POSITIONAL_OR_KEYWORD),
+                Parameter(
+                    "return_only_scores",
+                    Parameter.KEYWORD_ONLY,
+                    annotation=bool,
+                    default=False,
+                ),
+                *[
+                    Parameter(
+                        name,
+                        Parameter.KEYWORD_ONLY,
+                        annotation=allowed_kwargs_params[name].annotation,
+                    )
+                    for name in allowed_kwargs_params.keys()
+                ],
+            ]
+        )
+        call_method = _call_method
+
+    return type(
+        "CallableEvaluationMetric",
+        (EvaluationMetric,),
+        {"__call__": call_method},
+    )
+
+
+def _convert_val_to_pd_Series(val, name):
+    if val is not None and not isinstance(val, pd.Series):
+        if isinstance(val, str):
+            return pd.Series([val])
+        elif isinstance(val, list):
+            return pd.Series(val)
+        else:
+            raise TypeError(f"Expected {name} to be a string or list, got {type(val)}")
```

Comment:
```suggestion
            raise TypeError(f"Expected {name} to be a string, list, or Pandas Series, got {type(val)}")
```

nit: I think it's more friendly to list valid types to the `__call__` API, not this internal function

Reply:
Emm do you mean checking all passed value types directly in the `__call__` API? This is called inside the `__call__` API, and the parameter annotation uses `Union[pd.Series, str, list]`

Reply:
Nah I just mean add `Pandas Series` here because it is a valid input format.

Reply:
OH lol definitely!